packages/local_auth/local_auth_darwin/darwin/local_auth_darwin/Sources/local_auth_darwin/SystemWrappers.swift

```diff
@@ -0,0 +1,106 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import LocalAuthentication
+
+#if os(macOS)
+  import Cocoa
+  import FlutterMacOS
+#elseif os(iOS)
+  import Flutter
+  import UIKit
+#endif
+
+/// Protocol for interacting with LAContext instances, abstracted to allow using mock/fake instances
+/// in unit tests.
+protocol AuthContext {
+  var localizedFallbackTitle: String? { get set }
+  var biometryType: LABiometryType { get }
+  func canEvaluatePolicy(
+    _ policy: LAPolicy,
+    error: NSErrorPointer
+  ) -> Bool
+  func evaluatePolicy(
+    _ policy: LAPolicy,
+    localizedReason: String,
+    reply: @escaping @Sendable (Bool, Error?) -> Void
+  )
+}
+
+/// AuthContext is intentionally a direct passthroguh to LAContext.
+extension LAContext: AuthContext {}
+
+/// Protocol for a source of AuthContext instances. Used to allow context injection in unit
+/// tests.
+protocol AuthContextFactory {
+  func createAuthContext() -> AuthContext
+}
+
+// MARK: -
+
+#if os(macOS)
+  /// Protocol for interacting with NSAlert instances, abstracted to allow using mock/fake instances
+  /// in unit tests.
+  protocol AuthAlert {
+    @MainActor
+    var messageText: String { get set }
+    @MainActor
+    @discardableResult func addButton(withTitle title: String) -> NSButton
+    @MainActor
+    func beginSheetModal(
+      for sheetWindow: NSWindow,
+      completionHandler handler: ((NSApplication.ModalResponse) -> Void)?
+    )
+    @MainActor
```

Comment:
we can mark the whole protocol `@MainActor` so no need each function. (same below)